packages/blocks/src/store/selectors.js

```diff
@@ -95,6 +95,38 @@ export function getBlockVariations( state, blockName, scope ) {
 	} );
 }
 
+/**
+ * Returns the active block variation for a given block based on its attributes.
+ *
+ * @param {Object}                state      Data state.
+ * @param {string}                blockName  Name of block (example: “core/columns”).
+ * @param {Object}                attributes Block attributes used to determine active variation.
+ * @param {WPBlockVariationScope} [scope]    Block variation scope name.
+ *
+ * @return {(WPBlockVariation|void)} Active block variation.
```

Comment:
I think `void` is incorrect here. It's probably `undefined` when the matching active variation is not find.

Reply:
Fixed in https://github.com/WordPress/gutenberg/pull/30913/commits/66af876aef4138faad1fd07762dd5a26a6495b88